packages/@lwc/synthetic-shadow/src/env/shadow-root.ts

```diff
@@ -1,17 +1,12 @@
 /*
- * Copyright (c) 2018, salesforce.com, inc.
+ * Copyright (c) 2023, Salesforce.com, inc.
  * All rights reserved.
  * SPDX-License-Identifier: MIT
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
  */
 import { isNull } from '@lwc/shared';
 
-let NativeShadowRoot: any = null;
-if (typeof ShadowRoot !== 'undefined') {
-    NativeShadowRoot = ShadowRoot;
-}
-
-export const isNativeShadowRootDefined = !isNull(NativeShadowRoot);
+const NativeShadowRoot = ShadowRoot;
```

Comment:
nit: do we still need `NativeShadowRoot` or can that be removed and just have `isInstanceOfNativeShadowRoot` reference `ShadowRoot` directly? It's only referenced twice in this file and not exported.

Reply:
I think we still need to capture the `globalThis.ShadowRoot` once, since synthetic shadow will override it. But I can definitely clean up this logic: 766bfab8e